language/compiler/ir-to-bytecode/src/compiler.rs

```diff
@@ -1097,6 +1115,14 @@ fn compile_expression(
                     push_instr!(exp.span, Bytecode::Xor);
                     vec_deque![InferredType::U64]
                 }
+                BinOp::Shl => {
```

Comment:
is this code compiling always assuming u64? I am a bit confused by the fact there is no mention of the other integers here

Reply:
Looks like a bug to me. Good catch.